src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -50,60 +50,142 @@ std::string SpatialJoinAlgorithms::betweenQuotes(
   }
 }
 
-struct DistanceVisitor : public boost::static_visitor<double> {
-  template <typename Geometry1, typename Geometry2>
-  double operator()(const Geometry1& geom1, const Geometry2& geom2) const {
-    return bg::distance(geom1, geom2);
-  }
+bool SpatialJoinAlgorithms::getAnyGeometry(const IdTable* idtable, size_t row,
+                                           size_t col,
+                                           AnyGeometry& geometry) const {
+  auto printWarning = [alreadyWarned = false,
+                       &spatialJoin = spatialJoin_]() mutable {
+    if (!alreadyWarned) {
+      std::string warning =
+          "The input to a spatial join contained at least one element, "
+          "that is not a point or polygon geometry and is thus skipped. Note "
+          "that QLever currently only accepts point or polygon geometries for "
+          "the spatial joins";
+      AD_LOG_WARN << warning << std::endl;
+      alreadyWarned = true;
+      if (spatialJoin.has_value()) {
+        AD_CORRECTNESS_CHECK(spatialJoin.value() != nullptr);
+        spatialJoin.value()->addWarning(warning);
+      }
+    }
+  };
+
+  std::string str(betweenQuotes(ExportQueryExecutionTrees::idToStringAndType(
```

Comment:
Add a comment that explains that we need a fully materialized string for `bg::read_wkt` (which is somewhat unfortunate).